library/core/src/ptr/mod.rs

```diff
@@ -954,6 +954,23 @@ pub const unsafe fn swap<T>(x: *mut T, y: *mut T) {
     }
 }
 
+/// Non-overlapping *typed* swap of a single value.
+///
+/// The codegen backends will replace this with a better implementation when
+/// `T` is a simple type that can be an immediate.
+///
+/// # Safety
+///
+/// `x` and `y` are readable and writable as `T`, and non-overlapping.
+#[rustc_nounwind]
+#[inline]
+#[cfg_attr(not(bootstrap), rustc_intrinsic)]
```

Comment:
So far we've stuck to keeping all intrinsics in `core::intrinsics`. I think that's a good idea as intrinsics are fundamental primitives of the language, it's good to have them all in one place.

So I'd prefer if this function was declared in that file.